packages/flutter/lib/src/material/tabs.dart

```diff
@@ -34,23 +34,35 @@ const double _kMaxTabWidth = 264.0;
 ///  * [TabController], which coordinates tab selection between a [TabBar] and a [TabBarView].
 ///  * <https://material.google.com/components/tabs.html>
 class Tab extends StatelessWidget {
-  /// Creates a material design [TabBar] tab. At least one of [text] and [icon]
-  /// must be non-null.
+  /// Creates a material design [TabBar] tab. At least one of [text], [icon],
+  /// and [child] must be non-null. [text] and [child] cannot ne used at the
+  /// same time.
   const Tab({
     Key key,
     this.text,
     this.icon,
-  }) : assert(text != null || icon != null),
+    this.child,
+  }) : assert(text != null || child != null || icon != null),
+       assert(!(text != null && null != child)),
```

Comment:
Expand this assert to
assert(text == null || child == null);

Reply:
I wish I could. It's unfortunately an analyzer error: https://github.com/dart-lang/sdk/issues/31140

Reply:
Oh right, hence the discussion on Hackers.  Well, that sucks.  Maybe add a TODO with the bug reference...

Reply:
Done.